### DISCUSSION TITLE
Converting .JPG to .PNG makes image darker

### DISCUSSION THREAD

**Discussion by viddeyoyo** [ImageMagick/ImageMagick#4409] [Digital Image Processing] [answered]
When we are converting pictures  in our PIM using ImageMagick from .jpg to .png the pictures gamma is lowered resulting in a dark picture.

I've tried using both sRGB and RGB colorspace but the result is still the same. 

Down below I've attached the script that we've tried but they all make the .png darker. Has anyone had the same problem?

![image](https://user-images.githubusercontent.com/93139740/138709943-9676dd42-0aca-4eba-8340-5d8ca37491f7.png)
![image](https://user-images.githubusercontent.com/93139740/138709995-64aaf3c2-1759-4427-8339-d35999e9feda.png)

![image](https://user-images.githubusercontent.com/93139740/138710464-27806896-bf99-460a-85af-97df390ab62e.png)

Edit: I can see that the difference does not show when I uploaded the pictures. Hopefully the screenshot helps you see the difference.

![YQM317__01##monark karla emerald WebJpg](https://user-images.githubusercontent.com/93139740/138709519-154130b5-77b5-457b-809b-e774df073d0b.jpg)
![YQM317__01##monark karla emerald WebPng](https://user-images.githubusercontent.com/93139740/138709533-f751edf9-4b66-4733-8558-e08925217bd0.png)

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-colorspace RGB" converts to linear RGB. I suspect you really want (non-linear) sRGB, so don't convert to RGB.

You seem to be saying that the images themselves are fine, but the thumbnails shown in your screenshots are not. If so, you can remove the thumbnails with "-strip", but this will also remove the clipping path.

A more targeted method uses exiftool to remove all tags that start with "Thumb":
```
exiftool -Thumb*= myfile.jpg
```

**Reply by viddeyoyo** [reply]
We use it on the latest Inriver PIM platform. Unfortunately I don't know what version of IM they use. I'm using Windows 10 Pro 19042.1288. 

Ok, thanks for the input. But the sRGB version has the same problem. 

It's not only the thumbnails, if I open the picture in Photoshop the dark shade is present. It's only when I open the file in windows imageviewer it seems do adjust the gamma when opening the file.

**Reply by fmw42** [reply]
Perhaps you need to add an sRGB profile to the output.  Also note my comments about the proper syntax order. That would be more important in IM 7 than IM 6. So be sure you know which version of IM you are using.

**Reply by viddeyoyo** [reply]
I'll make the changes in the syntax order and see if it helps. Thanks for the help!

**Comment by fmw42** [comment, marked as answer]
Always read your raster input images before processing. So this works fine for me on IM 6.9.12.25 Q16 Mac OSX.

`convert bicycle.jpg -type truecolor -quality 75 -colorspace srgb -density 75 -resize 1500x1500 x.png`

However, it would be better to convert to srgb (though better to use profiles), then resize, then provide the output settings.

`convert bicycle.jpg -colorspace srgb -resize 1500x1500 -type truecolor -quality 75 -density 75  x.png`

![bicycle](https://user-images.githubusercontent.com/13458218/138718152-7d28aae7-2a67-4081-ab0b-12a3038dd620.jpg)

**What version of IM are you using and on what platform?**

**Reply by viddeyoyo** [reply]
Thanks everyone for the help! The solution was to put the commands in the proper syntax order as @fmw42 suggested. Now it works perfectly. Thanks again everyone for the help!
